server/controllers/user.js

```diff
@@ -200,4 +200,53 @@ module.exports = {
       return res.status(500).send("Internal Server Error");
     }
   },
+  async updateReviewAndUserRating(req, res) {
+    const { userId } = req.user;
+    const { reviewId, rating } = req.body;
+
+    // Updates Target Review Rating
+    try {
+      const request = await Request.findOne({ "embeddedReview._id": reviewId });
+
+      if (rating >= 1 && rating <= 5) {
+        request.embeddedReview.rating = rating;
+      } else {
+        throw new Error(
+          "Rating provided is not either in an acceptable format or is not a value within 1 to 5."
+        );
+      }
+
+      await request.save();
+      return res.status(201).send(request.toObject());
```

Comment:
should put logic inside one try and catch block since the code will not go below here once it is return

Reply:
I will check on this as this worked when I tested it.